Microsoft.Toolkit.Services/Services/LinkedIn/LinkedInDataProvider.cs

```diff
@@ -78,11 +81,28 @@ public LinkedInDataProvider(LinkedInOAuthTokens tokens, LinkedInPermissions requ
         /// <param name="requiredPermissions">Required permissions for the session.</param>
         public LinkedInDataProvider(LinkedInOAuthTokens tokens, LinkedInPermissions requiredPermissions)
         {
-            Tokens = tokens;
+            Tokens = tokens ?? throw new ArgumentException("Invalid token");
+            RequiredPermissions = requiredPermissions;
+            _authentication = new UwpAuthenticationBroker() ?? throw new ArgumentException("Invalid AuthenticationBroker");
```

Comment:
This is weird. You just instantiated the object. It can't be null. It will never throw. You need to validate only the parameters from this method.